File: LinkList/InsertAtTail.c
SinglyLinkedListNode* insertNodeAtTail(SinglyLinkedListNode* head, int data) {
    struct SinglyLinkedListNode *temp1,*temp;
    temp1 = (struct SinglyLinkedListNode*)malloc(sizeof(SinglyLinkedListNode));
    temp->data = data;
    temp1->next = NULL;
    if(head == NULL){
        head = temp1;
    }
    else{
        temp = head;
        while(temp->next != NULL){
            temp=temp->next;
        }
        temp->next = temp1;
    }
    return head;
}

File: LinkList/insert_at_tail.py
def insertNodeAtTail(head, data):
    temp = SinglyLinkedList()
    temp.data = data
    temp.next = None
    if head == None:
        head = temp
        return head
    else:
        temp1 = head
        while(temp1.next != None):
            temp1=temp1.next
        temp1.next = temp
        return head
